pkg/apis/core/validation/shoot.go

```diff
@@ -1734,26 +1736,62 @@ func ValidateWorkers(workers []core.Worker, fldPath *field.Path) field.ErrorList
 		workerNames.Insert(worker.Name)
 	}
 
-	allErrs = append(allErrs, ValidateSystemComponentWorkers(workers, fldPath)...)
-
 	return allErrs
 }
 
 // ValidateSystemComponentWorkers validates workers specified to run system components.
-func ValidateSystemComponentWorkers(workers []core.Worker, fldPath *field.Path) field.ErrorList {
+func ValidateSystemComponentWorkers(kubernetesVersion string, workers []core.Worker, fldPath *field.Path) field.ErrorList {
 	var (
 		allErrs                                   = field.ErrorList{}
 		atLeastOnePoolWithAllowedSystemComponents = false
+
+		workerPoolsWithSufficientWorkers   = make(map[string]struct{})
+		workerPoolsWithInsufficientWorkers = make(map[string]int)
 	)
 
-	for _, worker := range workers {
+	for i, worker := range workers {
+		// check if system component worker pool is configured
 		if !helper.SystemComponentsAllowed(&worker) {
 			continue
 		}
+
 		if worker.Minimum == 0 || worker.Maximum == 0 {
 			continue
 		}
 		atLeastOnePoolWithAllowedSystemComponents = true
+
+		// Check if maximum workers is at least as big as there are zones specified.
+		// It ensures, that the cluster potentially has one worker per zone to schedule required system components with TopologySpreadConstraints.
+		// This is considered per distinct worker pool concerning their zone setup,
+		// e.g. worker.zones: {1,2,3} == worker.zones: {3,2,1}
```

Comment:
```suggestion
		// Check if the maximum worker count is at least as high as the number of specified zones.
		// It ensures that the cluster has at least one worker node per zone in order to schedule required system components with TopologySpreadConstraints.
		// This is considered per distinct worker pool concerning their zone setup,
		// e.g. worker.zones: {1,2,3} == worker.zones: {3,2,1}
```

Reply:
What does 

```
		// This is considered per distinct worker pool concerning their zone setup,
		// e.g. worker.zones: {1,2,3} == worker.zones: {3,2,1}
```

mean? 👀

Reply:
I means that for the check, worker pool 'x' with zones {1,2,3} is considered the same as pool 'y' with zones {3,2,1}. I adapted the comment to point this out.